### PR TITLE
Rough draft of RFC-5000: Payment channels

### LINKED ISSUE #99: RFC-0500 Payment Channels
This RFC will describe how Tari implements payment channels. 

The base layer is slow. The DAN is fast. 

However, every non-read DAN instruction will probably have a fee (i.e. base layer transaction) associated with it. To bridge the speed gap between the two layers, a payment channel solution is required. This document provides the high-level overview of how this is done.

### PR DESCRIPTION
**Pull request by CjS77**
## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [x] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch
* [ ] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

Closes #99

### CODE REVIEW

File: RFC/src/RFC-0500_PaymentChannels.md [hunk 1]
```diff
@@ -1,8 +1,91 @@
...
+  committee.
+* Users can request a withdrawal for their Tari at any time. At predefined intervals, the Clacks committee will process
+  those withdrawal requests and give the Tari UTXO control back to the user.
+
+Users also have a pre-signed, time-locked transaction that returns the user's fund back to them. This refund transaction
+can be used in case the Clacks committee stops processing withdrawals and provides insurance for users against having
+their funds locked up forever.
+
+The peg-in, peg-out cycle are represented by standard Tari transactions. This means that full nodes verify that no funds
+have been created or destroyed over the course of the cycle. They do not check that the "balances" associated with users
```
Comment (hansieodendaal) [inline RFC/src/RFC-0500_PaymentChannels.md]:
```suggestion
have been created or destroyed over the course of the cycle. With level 0 security, they do not check that the "balances" associated with users
```

File: RFC/src/RFC-0500_PaymentChannels.md [hunk 2]
```diff
@@ -1,8 +1,91 @@
...
+are what those users would expect in an honestly run side chain.
+
+However, if the side-chain is run as a standard mimblewimble process, any third party running a full node and that
+access to the opening and closing channel balances, can in fact verify that the committee has operated honestly.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0500_PaymentChannels.md]:
```suggestion
has access to the opening and closing channel balances, can, in fact, verify that the committee has operated honestly.
```

File: RFC/src/RFC-0500_PaymentChannels.md [hunk 3]
```diff
@@ -1,8 +1,91 @@
...
+
+Users also have a pre-signed, time-locked transaction that returns the user's fund back to them. This refund transaction
+can be used in case the Clacks committee stops processing withdrawals and provides insurance for users against having
+their funds locked up forever.
+
+The peg-in, peg-out cycle are represented by standard Tari transactions. This means that full nodes verify that no funds
+have been created or destroyed over the course of the cycle. They do not check that the "balances" associated with users
+are what those users would expect in an honestly run side chain.
+
+However, if the side-chain is run as a standard mimblewimble process, any third party running a full node and that
```
Comment (hansieodendaal) [inline RFC/src/RFC-0500_PaymentChannels.md]:
```suggestion
However, if the side-chain is run as a standard Mimblewimble process (i.e. security levels 1 or 2), any third party running a full node and that
```

File: RFC/src/RFC-0500_PaymentChannels.md [hunk 4]
```diff
@@ -1,8 +1,91 @@
...
+## Related Requests for Comment
+
+* [RFC-0001: Overview](RFC-0001_overview.md)
+* [RFC-0300: The Tari Digital Assets Network](RFC-0300_DAN.md)
+
+## Description
+
+### Introduction
+
+The base layer is slow. The DAN is fast. However, every non-read DAN instruction has a fee (i.e. base layer transaction)
```
[the remark this answers is not in the document]
Comment (CjS77) [reply RFC/src/RFC-0500_PaymentChannels.md]: non-read = create, modify or delete.

[the remark this answers is not in the document]
Comment (CjS77) [reply RFC/src/RFC-0500_PaymentChannels.md]: Text has been updated to be more explicit.

**Pull request merged**